packages/attention/src/component.tsx

```diff
@@ -91,47 +100,52 @@ export function Attention(props: AttentionProps) {
       )}
       ref={attentionRef}
     >
-      <div
-        className={classNames({
-          [c.base]: true,
-          [c.tooltip]: props.tooltip,
-          [c.callout]: props.callout,
-          [c.popover]: props.popover,
-        })}
-      >
+      <div className={wrapperClasses}>
         {!props.noArrow && (
           <Arrow {...props} ref={arrowRef} direction={placement} />
         )}
-        <div className="last-child:mb-0">{props.children}</div>
+        <div className="last:mb-0">{props.children}</div>
       </div>
     </div>
   );
 }
 
-const Arrow = forwardRef<HTMLDivElement, ArrowProps>((props, ref) => {
-  const { direction, tooltip, callout, popover } = props;
+const arrowDirectionClassname = (dir: string) => {
+  const direction = dir.charAt(0).toUpperCase() + dir.slice(1);
+
+  return `arrowDirection${direction}`;
+}
 
+const Arrow = forwardRef<HTMLDivElement, ArrowProps>((props, ref) => {
+  const { callout, direction, popover, tooltip } = props;
   const arrowDirection = opposites[direction];
 
+  const arrowType = () => {
+    if (tooltip) return 'arrowTooltip';
+    else if (callout) return 'arrowCallout';
+    else if (popover) return 'arrowPopover';
+    return '';
+  }
+
+  const arrowClasses = classNames(
+    ccAttention.arrowBase,
+    ccAttention[arrowDirectionClassname(arrowDirection)],
+    {
+      [ccAttention[arrowType()]]: tooltip || callout || popover
```

Comment:
Imho it was much more readable before when there were no need for a function to jump into
``` 
   [ccAttention.tooltip]: props.tooltip,
   [ccAttention.callout]: props.callout,
   [ccAttention.popover]: props.popover,
```

If the refactor is necessary I would maybe suggest to add an enum or something and skip the function